test/jdk/java/nio/file/Files/probeContentType/Basic.java

```diff
@@ -36,13 +37,25 @@
 import java.util.List;
 import java.util.stream.Stream;
 
+import jdk.internal.util.OperatingSystem;
+import jdk.internal.util.OSVersion;
+
 /**
  * Uses Files.probeContentType to probe html file, custom file type, and minimal
  * set of file extension to content type mappings.
  */
 public class Basic {
-    private static final boolean IS_UNIX =
-        ! System.getProperty("os.name").startsWith("Windows");
+    // Whether the OS is Windows with version 11 or higher
+    private static final boolean IS_WIN11_PLUS = OperatingSystem.isWindows() &&
+        (System.getProperty("os.name").endsWith("11") ||
+         new OSVersion(10, 0).compareTo(OSVersion.current()) > 0);
+
+    // Extra MIME types expected by Windows 11+
+    private static final String EXTRA_BZ2 = IS_WIN11_PLUS ? "application/x-compressed" : "";
+    private static final String EXTRA_CSV = IS_WIN11_PLUS ? "application/vnd.ms-excel" : "";
+    private static final String EXTRA_RAR = IS_WIN11_PLUS ? "application/x-compressed" : "";
+    private static final String EXTRA_RTF = IS_WIN11_PLUS ? "application/msword" : "";
+    private static final String EXTRA_7Z  = IS_WIN11_PLUS ? "application/x-compressed" : "";
```

Comment:
> Does WIndows 11 have a (user) way to change the mapping? 

The mapping comes from the Windows registry so it could be changed but I wouldn't expect that to be common. Part of me wonders if this test should be reduced to a small number of extensions, 30 seems a lot to have to keep up to date with changes like this.

Reply:
Not related to the above comments, but it looks as if the Windows registry might only be able to handle one MIME type per extension. Also, I wonder how much other software was broken by this mapping change.

Reply:
> It could be misleading to put a empty string into the expected set. Could the lists be created depending on the OS?

I agree. I'll see what I can change.

> Does WIndows 11 have a (user) way to change the mapping? For example, the mapping of .csv to vnd.ms-excel might not be correct if the user selected OpenOffice to handle .csv. I don't think we have a mechanism to cope with that.

No, we don't.